drools-wb-services/drools-wb-verifier/drools-wb-verifier-web-worker/src/main/java/org/drools/workbench/verifier/webworker/client/Receiver.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.drools.workbench.verifier.webworker.client;
+
+import java.util.Set;
+
+import org.drools.workbench.services.verifier.api.client.Status;
+import org.drools.workbench.services.verifier.api.client.configuration.AnalyzerConfiguration;
+import org.drools.workbench.services.verifier.api.client.configuration.RunnerType;
+import org.drools.workbench.services.verifier.api.client.index.Index;
+import org.drools.workbench.services.verifier.api.client.reporting.Issue;
+import org.drools.workbench.services.verifier.api.client.reporting.Issues;
+import org.drools.workbench.services.verifier.api.client.util.PortablePreconditions;
+import org.drools.workbench.services.verifier.core.main.Analyzer;
+import org.drools.workbench.services.verifier.core.main.Reporter;
+import org.drools.workbench.services.verifier.plugin.client.api.DeleteColumns;
+import org.drools.workbench.services.verifier.plugin.client.api.Initialize;
+import org.drools.workbench.services.verifier.plugin.client.api.MakeRule;
+import org.drools.workbench.services.verifier.plugin.client.api.NewColumn;
+import org.drools.workbench.services.verifier.plugin.client.api.RemoveRule;
+import org.drools.workbench.services.verifier.plugin.client.api.RequestStatus;
+import org.drools.workbench.services.verifier.plugin.client.api.Update;
+import org.drools.workbench.services.verifier.plugin.client.api.WebWorkerException;
+import org.drools.workbench.services.verifier.plugin.client.builders.BuildException;
+
+public class Receiver {
+
+    private Analyzer analyzer;
+    private Poster poster;
+    private RunnerType runnerType;
+    private Issues latestReport;
+    private Index index;
+    private AnalyzerConfiguration configuration;
+
+    public Receiver( final Poster poster,
+                     final RunnerType runnerType ) {
+        this.poster = PortablePreconditions.checkNotNull( "poster",
+                                                          poster );
+        this.runnerType = PortablePreconditions.checkNotNull( "runnerType",
+                                                              runnerType );
+    }
+
+    public void received( final Object o ) {
+
+        if ( o instanceof Initialize ) {
```

Comment:
Yep. Thought about making it possible to register a listeners for different types of objects/commands, but this list would just be a list of added listeners. And since this is the only setup that is ever used I felt instanceof + casting was enough for now, but if Errai for example wants to extend the Web Worker support, I could add that there and then we could reuse it. Even callbacks would be possible. 

I almost did callbacks, but then took the easier and less chatty way. This was when I still hoped to keep creating the Index in the main application. (Might move to server side later and call server side directly from the web worker)